Migration to JDA:v5.0.0-beta

jocull opened the pull request:
As requested in #65 

I followed the migration guide here https://jda.wiki/introduction/migration-v4-v5/ I hope I did this right 🙂

jocull committed:
Rough migration to JDA:v5.0.0-beta.6

Not sure I did everything exactly right. A migration guide is covered
here, but I did most of this in a hurry.

https://jda.wiki/introduction/migration-v4-v5/

jocull committed:
Bump to JDA 5 beta 13

jocull committed:
Remove comment that doesn't apply anymore

jocull committed:
Bump to dev.arbjerg:lavaplayer:2.0.1

jocull committed:
Bump to JDA 5.0.0-beta.20

jocull committed:
Bump to lavaplayer:2.1.0

jocull committed:
Use a commit version right after 2.1.0 (727959e9f6)

See https://github.com/lavalink-devs/lavaplayer/pull/70
And https://jitpack.io/#lavalink-devs/lavaplayer/727959e9f6

Frozen-byte commented:
Would you mind releasing a docker for this? I would love to test it then 👍🏻

jocull committed:
Bump lavaplayer to 2.1.1 and stop using a commit hash

See
- https://github.com/lavalink-devs/lavaplayer/pull/70
- https://github.com/lavalink-devs/lavaplayer/releases/tag/2.1.1

jocull commented:
Hey @Frozen-byte 👋 😄 Please try https://github.com/jocull/ukulele/pkgs/container/ukulele/192687440?tag=jda-5
I hope I did this right!

Frozen-byte commented:
> Hey @Frozen-byte 👋 😄 Please try https://github.com/jocull/ukulele/pkgs/container/ukulele/192687440?tag=jda-5 I hope I did this right!

Works perfectly, thank you! Will test it further this week and report back if any odd behavior was noticed.

jocull committed:
Use snapshot for com.github.Nansess.lavaplayer-test:lavaplayer:19e815bd83

This fixes upstream issue with lavaplayer for error:
"Video returned by YouTube isn't what was requested"

See https://github.com/lavalink-devs/lavaplayer/pull/93

jocull commented:
The image for this was updated to address new issue https://github.com/freyacodes/ukulele/issues/73#issuecomment-2030886146

See https://github.com/jocull/ukulele/pkgs/container/ukulele/199356264?tag=jda-5

jocull committed:
Changes to use new YoutubeAudioSourceManager

jocull committed:
Bump JDA to latest, beta 5.23

Frozen-byte commented:
Sadly today it got borked again, look like alphabet getting serious :/
image: ghcr.io/jocull/ukulele:jda-5

Message:
```
ukulele-1  | 2024-04-29 17:04:36.399 ERROR 1 --- [ader-2-thread-1] dev.arbjerg.ukulele.command.PlayCommand  : Handled exception occurred
ukulele-1  | 
ukulele-1  | com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Video returned by YouTube isn't what was requested
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.DefaultYoutubeTrackDetailsLoader.load(DefaultYoutubeTrackDetailsLoader.java:59) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.DefaultYoutubeTrackDetailsLoader.loadDetails(DefaultYoutubeTrackDetailsLoader.java:36) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadTrackWithVideoId(YoutubeAudioSourceManager.java:230) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager$LoadingRoutes.track(YoutubeAudioSourceManager.java:254) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager$LoadingRoutes.track(YoutubeAudioSourceManager.java:250) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.DefaultYoutubeLinkRouter.routeFromUrlWithVideoId(DefaultYoutubeLinkRouter.java:103) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.DefaultYoutubeLinkRouter.routeFromMainDomain(DefaultYoutubeLinkRouter.java:68) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.DefaultYoutubeLinkRouter.route(DefaultYoutubeLinkRouter.java:46) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadItemOnce(YoutubeAudioSourceManager.java:218) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadItem(YoutubeAudioSourceManager.java:149) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.checkSourcesForItemOnce(DefaultAudioPlayerManager.java:412) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.checkSourcesForItem(DefaultAudioPlayerManager.java:394) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.loadItemSync(DefaultAudioPlayerManager.java:156) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$loadItem$0(DefaultAudioPlayerManager.java:175) ~[lavaplayer-19e815bd83.jar!/:na]
ukulele-1  |    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
ukulele-1  |    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
ukulele-1  |    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
ukulele-1  |    at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
ukulele-1  | Caused by: java.lang.IllegalStateException: 
```

jocull commented:
Some of the included libraries in this project have shifted around, I do have an updated branch that was working, but have not used it in a few days. I can attempt to validate and push updates again.

jocull committed:
Fix remaining quirks with YoutubeAudioSourceManager to exclude old version

jocull committed:
Clean up imports

jocull commented:
Hi @Frozen-byte - I've updated the branch and Docker image. You can give the updates a try and see if they make an improvement for you 😃

jocull committed:
Update versions and clean up audio source managers

jocull committed:
Use a version of JDK 11 that isn't deprecated

jocull committed:
Use the version expected in the docker-compose for ease of use

jocull committed:
Use the current Dockerfile as the image

jocull committed:
Make the container name predictable

Frozen-byte commented:
I am sorry for the late testing, was on vacation.
Works like a charm, I had one minor caching issue and needed to `docker down` the container instead of just pulling the new image.

jocull committed:
Manually add jars for youtube-plugin and upgrade version

From https://github.com/lavalink-devs/youtube-source/releases/tag/1.4.0
The jitpack is currently failing at https://jitpack.io/#lavalink-devs/lavaplayer-youtube-source

huntastikus commented:
@jocull I am getting `com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Sign in to confirm you’re not a bot
`

jocull commented:
> @jocull I am getting `com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Sign in to confirm you’re not a bot `

There are some upstream PRs I’m trying to monitor for this, but unsure of a solution currently. Thanks for the report, I will update when there’s something more clear 🙏

Frozen-byte commented:
> > @jocull I am getting `com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Sign in to confirm you’re not a bot `
> 
> There are some upstream PRs I’m trying to monitor for this, but unsure of a solution currently. Thanks for the report, I will update when there’s something more clear 🙏

is there something we can help with? I have this problem, too.

jocull committed:
Bump youtube-plugin to 1.5.2

From https://github.com/lavalink-devs/youtube-source/releases/tag/1.5.2

jocull commented:
@huntastikus @Frozen-byte the branch and docker image are updated with a newer version. It seems to work for me, but don't know if it will solve your specific issue. Let me know your results please :)

kiloforce commented:
> image: ghcr.io/jocull/ukulele:jda-5

Back work working again, thanks @jocull !

kiloforce commented:
@jocull looks like you have had this PR up since April with no movement.  It looks like none of the other PRs have had any movement since early 2022?  Have you considered maybe upgrading your fork to a more full featured version?  Maybe one with an updated README / Wiki with instructions for bringing up Ukulele with your Docker?

huntastikus commented:
I am not sure how to apply the upstream work to this package. @kiloforce how did you get it to work?

kiloforce commented:
> I am not sure how to apply the upstream work to this package. @kiloforce how did you get it to work?

```
# original configuration
git clone https://github.com/freyacodes/ukulele.git
cd ukulele
git checkout master
git pull
cp ukulele.example.yml ukulele.yml
# configure ukulele.yml
mkdir db && chown -R 999 db/
---

# routinely update jda-5 docker image:
docker rm -f ukulele_ukulele_1
docker pull ghcr.io/jocull/ukulele:jda-5
docker-compose up -d

# follow the logs to make sure it is happy:
docker logs -f ukulele_ukulele_1
```

Frozen-byte commented:
> @huntastikus @Frozen-byte the branch and docker image are updated with a newer version. It seems to work for me, but don't know if it will solve your specific issue. Let me know your results please 😄
> 
> If this doesn't work, then I believe this upstream work is likely relevant: [lavalink-devs/youtube-source#33](https://github.com/lavalink-devs/youtube-source/pull/33)

Since my server is in the banned IP-Range of google, it won't work until https://github.com/lavalink-devs/youtube-source/pull/33 hits this repository I think. Lavalink should provide this fix in the next version, and I will happily test it when it arrives ukulele, too!

jocull committed:
Swap to youtube-v2 1.8.3 source, drop manual jars

jocull committed:
Set the docker-compose to use the image by default

jocull commented:
I'll post an important update in #77 with where I feel the situation is for this. It's not as straightforward to fix, and there is little consensus upstream on what works and doesn't.

freyacodes reviewed (commented):
Hi! Sorry for never getting back to this until now!

With FredBoat shutting down, I would like to take the time to revive this project. A V2 if you will. This branch is also in an outdated state, but it is good enough as a base for continued work. Thank you!

I'm also happy to collaborate with anyone wishing to help maintain the project going forward so that it can continue living independently of me.
